lib/base/application.cpp

```diff
@@ -180,7 +180,7 @@ void Application::SetResourceLimits(void)
 	rlimit rl;
 
 #	ifdef RLIMIT_NOFILE
-	rl.rlim_cur = 16 * 1024;
+	rl.rlim_cur = GetRLimitFiles();
```

Comment:
The patch sets the lower limits already for Linux, and allows the user to override this with their own values in the future. I wouldn't change this behaviour to make it "full user controllable", then the benefit of default values is entirely gone and users might run into issues again.
Keep in mind that init.conf is a configuration file, and we cannot override this for 2.7.